File: web/frontend/src/types/jobs.ts
export type StageStatus = "idle" | "running" | "done" | "failed" | "skipped";
export type RuntimeStatus = "queued" | "running" | "succeeded" | "failed" | "idle";

export interface StageSummary {
  status: StageStatus;
  success: boolean | null;
  reused_existing?: boolean | null;
  error_code?: string | null;
  hint?: string | null;
}

export interface JobListItem {
  id: string;
  job_dir: string;
  title: string;
  source_url: string | null;
  source_path: string | null;
  command: string | null;
  updated_at: string | null;
  stages: Record<string, StageSummary>;
  runtime_status: RuntimeStatus;
}

export interface JobListResponse {
  jobs: JobListItem[];
  total: number;
}

export interface JobDetailResponse {
  id: string;
  job_dir: string;
  manifest: Record<string, unknown>;
  stages: Record<string, StageSummary>;
  runtime_status: RuntimeStatus;
  log?: string | null;
}

export interface DoctorCheck {
  name: string;
  ok: boolean;
  detail: string;
  section: string;
  code: string | null;
  hint: string | null;
}

export interface DoctorResponse {
  checks: DoctorCheck[];
  output_dir: string;
  config_source: string | null;
}

export interface CreateJobResponse {
  id: string;
  job_dir: string | null;
  runtime_status: RuntimeStatus;
  message: string;
}

export interface ArtifactPreview {
  artifact_key: string;
  kind: "text" | "markdown" | "json" | "subtitle" | "media";
  content: string | null;
  filename: string | null;
  media_url: string | null;
}

export type PreviewTabId = "transcript" | "summary" | "subtitle" | "media" | "keywords";

export const PREVIEW_TABS: { id: PreviewTabId; label: string; keys: string[] }[] = [
  { id: "transcript", label: "transcript", keys: ["transcript_txt"] },
  { id: "summary", label: "summary", keys: ["summary_md"] },
  { id: "subtitle", label: "subtitle", keys: ["subtitle_srt", "subtitle_vtt"] },
  { id: "media", label: "media", keys: ["video", "audio"] },
  { id: "keywords", label: "keywords", keys: ["keywords_json"] },
];

File: web/api/schemas/jobs.py
"""Job-related API schemas."""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, Field

StageStatus = Literal["idle", "running", "done", "failed", "skipped"]
RuntimeStatus = Literal["queued", "running", "succeeded", "failed", "idle"]


class StageSummary(BaseModel):
    status: StageStatus
    success: bool | None = None
    reused_existing: bool | None = None
    error_code: str | None = None
    hint: str | None = None


class JobListItem(BaseModel):
    id: str
    job_dir: str
    title: str
    source_url: str | None = None
    source_path: str | None = None
    command: str | None = None
    updated_at: str | None = None
    stages: dict[str, StageSummary]
    runtime_status: RuntimeStatus


class JobListResponse(BaseModel):
    jobs: list[JobListItem]
    total: int


class JobDetailResponse(BaseModel):
    id: str
    job_dir: str
    manifest: dict[str, Any]
    stages: dict[str, StageSummary]
    runtime_status: RuntimeStatus
    log: str | None = None


class WarningDetail(BaseModel):
    code: str | None = None
    stage: str | None = None
    message: str | None = None
    description: str | None = None


class CreateJobRequest(BaseModel):
    type: Literal["download", "download-subs", "transcribe", "summarize", "run"] = Field(
        description="Pipeline command type aligned with vlp CLI."
    )
    url: str | None = None
    input_path: str | None = None
    options: dict[str, Any] = Field(default_factory=dict)


class CreateJobResponse(BaseModel):
    id: str
    job_dir: str | None = None
    runtime_status: RuntimeStatus = "queued"
    message: str = "任务已加入队列"


class JobStatusResponse(BaseModel):
    id: str
    runtime_status: RuntimeStatus
    job_dir: str | None = None
    error: str | None = None
    error_code: str | None = None
    hint: str | None = None
    log: str | None = None
    stages: dict[str, StageSummary]


class ArtifactPreviewResponse(BaseModel):
    artifact_key: str
    kind: Literal["text", "markdown", "json", "subtitle", "media"]
    content: str | None = None
    filename: str | None = None
    media_url: str | None = None
